simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/handlers/ocpp/support/AvailabilityManager.java

```diff
@@ -0,0 +1,135 @@
+import com.evbox.everon.ocpp.v20.message.station.ChangeAvailabilityResponse;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.evbox.everon.ocpp.v20.message.station.ChangeAvailabilityResponse.Status.*;
+
+@Slf4j
+@AllArgsConstructor
+public class AvailabilityManager {
+
+    private final StationState stationState;
+    private final StationMessageSender stationMessageSender;
+
+    /**
+     * Change EVSE status by doing the following:
+     * <p>
+     * 1. Send response with ACCEPTED status when EVSE status is the same as requested.
+     * 2. Change EVSE status to the requested status when they do not match.
+     * In addition send response with ACCEPTED status and StatusNotification request for every EVSE Connector.
+     * 3. When a transaction is in progress.
+     * Send response with SCHEDULED status and save scheduled status for further processing.
+     * 4. Send response with REJECTED status when EVSE could not be found.
+     *
+     * @param callId              identity of the message
+     * @param request             incoming request from the server
+     * @param requestedEvseStatus requested EVSE status
+     */
+    public void changeEvseAvailability(String callId, ChangeAvailabilityRequest request, EvseStatus requestedEvseStatus) {
+
+        try {
+            Evse evse = stationState.findEvse(request.getEvseId());
+
+            ChangeAvailabilityResponse.Status evseStatus = defineEvseStatus(requestedEvseStatus, evse);
+
+            sendResponseWithStatus(callId, evseStatus);
+
+            sendNotificationRequest(evse);
+
+        } catch (IllegalArgumentException e) {
+            log.error(e.getMessage(), e);
+            sendResponseWithStatus(callId, REJECTED);
+        }
+
+    }
+
+    /**
+     *
+     * Change Station status by doing the following:
+     * <p>
+     * 1. Send response with ACCEPTED status when all EVSE statuses are the same as requested.
+     * 2. Change EVSE status to the requested status when they do not match.
+     * In addition send response with ACCEPTED status and StatusNotification request for every EVSE Connector.
+     * 3. When a transaction is in progress for at least one EVSE then send response with SCHEDULED status
+     * and save scheduled status for further processing.
+     * 4. Send response with REJECTED status when no EVSEs are present.
+     *
+     * @param callId              identity of the message
+     * @param requestedEvseStatus requested EVSE status
+     */
+    public void changeStationAvailability(String callId, EvseStatus requestedEvseStatus) {
+
+        if (stationState.getEvses().isEmpty()) {
+            sendResponseWithStatus(callId, REJECTED);
+            return;
+        }
+
+        ChangeAvailabilityResponse.Status stationStatus = defineStationStatus(requestedEvseStatus);
+
+        sendResponseWithStatus(callId, stationStatus);
+
+        for (Evse evse : stationState.getEvses()) {
```

Comment:
why not stationState.getEvses.forEach(this::sendNotificationRequest) ?

Reply:
Done